netmiko/nokia/nokia_sros_ssh.py

```diff
@@ -168,3 +170,95 @@ def cleanup(self, command="logout"):
         # Always try to send final 'logout'.
         self._session_log_fin = True
         self.write_channel(command + self.RETURN)
+
+
+class NokiaSrosFileTransfer(BaseFileTransfer):
+    def _get_cmd_prefix(self):
+        """
+        Returns "//" if the current prompt is MD-CLI
+        empty string otherwise
+        """
+        return "//" if "@" in self.ssh_ctl_chan.base_prompt else ""
+
+    def remote_space_available(self, search_pattern=r"(\d+)\s+\w+\s+free"):
+        """Return space available on remote device."""
+
+        # Sample text for search_pattern.
+        # "               3 Dir(s)               961531904 bytes free."
+        self.ssh_ctl_chan.send_command(self._get_cmd_prefix() + "environment no more")
+        remote_cmd = self._get_cmd_prefix() + "file dir {}".format(self.file_system)
+        remote_output = self.ssh_ctl_chan.send_command(remote_cmd)
+        match = re.search(search_pattern, remote_output)
+        return int(match.group(1))
+
+    def check_file_exists(self, remote_cmd=""):
+        """Check if destination file exists (returns boolean)."""
+
+        if self.direction == "put":
+            self.ssh_ctl_chan.send_command(
+                self._get_cmd_prefix() + "environment no more"
+            )
+            if not remote_cmd:
+                remote_cmd = self._get_cmd_prefix() + "file dir {}/{}".format(
+                    self.file_system, self.dest_file
+                )
+            remote_out = self.ssh_ctl_chan.send_command(remote_cmd)
+            if "File Not Found" in remote_out:
+                return False
+            elif self.dest_file in remote_out:
+                return True
+            else:
+                raise ValueError("Unexpected output from check_file_exists")
+        elif self.direction == "get":
+            return os.path.exists(self.dest_file)
+
+    def remote_file_size(self, remote_cmd=None, remote_file=None):
+        """Get the file size of the remote file."""
+
+        if remote_file is None:
+            if self.direction == "put":
+                remote_file = self.dest_file
+            elif self.direction == "get":
+                remote_file = self.source_file
+        if not remote_cmd:
+            remote_cmd = self._get_cmd_prefix() + "file dir {}/{}".format(
+                self.file_system, remote_file
+            )
+        self.ssh_ctl_chan.send_command(self._get_cmd_prefix() + "environment no more")
+        remote_out = self.ssh_ctl_chan.send_command(remote_cmd)
+
+        if "File Not Found" in remote_out:
+            raise IOError("Unable to find file on remote system")
+
+        # Parse dir output for filename. Output format is:
+        # "10/16/2019  10:00p                6738 {filename}"
+
+        pattern = r"(\S+)\s+(\S+)\s+(\d+)\s+{}".format(re.escape(remote_file))
```

Comment:
Is there a reason here for saving groups1 and groups2 here?

Reply:
fixed in 468b6d5